### PR TITLE
Generic PRNG framework:

### PR DESCRIPTION
**Pull request by nbougalis**
* A new, unified interface for generating random numbers and filling buffers
  supporting any engine that fits the UniformRandomNumberGenerator concept;
* Automatically seeded replacement for rand using fast xorshift+ PRNG engine;
* A CSPRNG engine that can be used with the new framework when needing to
  to generate cryptographically secure randomness.
* Unit test cleanups to work with new engine.

Reviews: @JoelKatz, @vinniefalco, @HowardHinnant, @scottschurr

### CODE REVIEW

**Comment by nbougalis** [comment]
Yep, silly me. I will fix it.

**Comment by scottschurr** [comment]
Thanks for the education.  I left a few comments that you can use (or not) at your discretion.  LGTM :+1:.

**Comment by HowardHinnant** [comment]
None of my comments were nitpicks or questions. :+1:

**Comment by nbougalis** [comment]
GitHub ate your comments guys, sorry. All of them have been addressed (in some fashion). Please take a second look at your convenience.

**Comment by JoelKatz** [comment]
:neckbeard::+1:

**Comment by scottschurr** [comment]
Looks good to me.  I left a couple of small opinions that may be taken or not.
:+1:

**Comment by vinniefalco** [comment]
nudb should not have dependencies on other beast sources, so it could be used stand-alone.

**Comment by vinniefalco** [comment]
The commit message is inappropriate for beast - it claims tests but there are no such tests in beast.

File: src/beast/beast/random/random.h [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <random>
+#include <limits>
+#include <type_traits>
+
+namespace beast {
+
+#if BEAST_CLANG
```
Comment (vinniefalco) [inline src/beast/beast/random/random.h]: Why?

**Pull request closed without merging**